modules/nix-darwin/templates/default.nix

```diff
@@ -49,18 +49,32 @@ in
                 '';
               };
               owner = mkOption {
-                type = types.singleLineStr;
-                default = "root";
+                type = with lib.types; nullOr singleLineStr;
+                default = null;
+                description = ''
+                  User of the file. Can only be set if uid is 0;
+                '';
+              };
+              uid = mkOption {
+                type = with lib.types; nullOr int;
+                default = 0;
                 description = ''
-                  User of the file.
+                  UID of the template, only applied with owner is null. the UID will be applied even if the corresponding user doesn't exist.
                 '';
               };
               group = mkOption {
-                type = types.singleLineStr;
-                default = "staff";
+                type = with lib.types; nullOr singleLineStr;
+                default = if config.owner != null then "staff" else null;
```

Comment:
I assume you have not tested this on macOS yet?
I just need to know if I need to give this a try before merging.

Reply:
That is correct, I do not have an OSX I can test it with.

Reply:
Also, just in case you didn't see it, I still haven't figured out how to add assertions, if you could offer a bit of advice there, I'd appreciate it :D

Reply:
Assertions should be fixed.